RealmSwift/SwiftUI.swift

```diff
@@ -645,6 +645,285 @@ extension EnvironmentValues {
+    }
+
+    /// :nodoc:
+    public var wrappedValue: AsyncOpenState {
+        storage.asyncOpenState
+    }
+
+    /**
+     This will cancel any notification from the property wrapper states
+     */
+    public func cancel() {
+        storage.cancellables.forEach { $0.cancel() }
+        storage.cancellables = []
+    }
+
+    /**
+     Initialise the property wrapper
+     - parameter appId: The unique identifier of your Realm app.
+     - parameter partitionValue: The `BSON` value the Realm is partitioned on.
+     - parameter baseConfiguration: A Base Configuration use when creating the Realm,
+     if empty the configuraion is setted to the `defaultConfiguration`
+     - parameter timeout: The maximum number of milliseconds to allow for a connection to
+     become fully established., if empty or `nil` no connection timeout is setted.
+     */
+    public init<T: BSON>(appId: String, partitionValue: T, baseConfiguration: Realm.Configuration = Realm.Configuration.defaultConfiguration, timeout: UInt? = nil) {
+        let app = App(id: appId)
+        if let timeout = timeout {
+            let syncTimeoutOptions = SyncTimeoutOptions()
+            syncTimeoutOptions.connectTimeout = timeout
+            app.syncManager.timeoutOptions = syncTimeoutOptions
+        }
+        if let currentUser = app.currentUser,
+           currentUser.isLoggedIn {
+            asyncOpenForUser(currentUser, partitionValue: partitionValue, configuration: baseConfiguration)
+        } else {
+            app.objectWillChange.sink { [self] app in
+                if let currentUser = app.currentUser {
+                    asyncOpenForUser(currentUser, partitionValue: partitionValue, configuration: baseConfiguration)
+                }
+            }.store(in: &storage.cancellables)
+        }
+    }
+
+    public mutating func update() {
+        if storage.realm == nil || storage.realm != realm {
+            storage.realm = realm
+        }
+    }
+}
+
+// MARK: - AutoOpen
+
+/// This property wrapper behaves similar as `AsyncOpen`, and in terms of declaration and use is completly identical,
+/// but with the difference of a offline-first approach.
+/// `AutoOpen` will try once to asynchronously open a Realm, but in case of no internet connection will return an opened realm
+/// for the given appId and partitionValue which can be used within our view.
+
+/// Add AutoOpen to your ``SwiftUI/View`` or ``SwiftUI/App``,  after a user is already logged,
+/// or if a user is going to be logged
+///
+///     @AutoOpen(appId: "app_id", partitionValue: <partition_value>, timeout: 4000) var autoOpen
+///
+/// This will inmeadiatly initiates a `Realm.asyncOpen()` operation which will perform all work needed to get the Realm to
+/// a usable state. (see Realm.asyncOpen() documentation)
+///
+/// This property wrapper will publish states of the current `Realm.asyncOpen()` process like progress, errors and an opened realm,
+/// which can be used to update the view
```

Comment:
There's several typos in here. Xcode has a built-in spellcheck for comments which you can enable at Edit -> Format -> Spelling and Grammar -> Check Spelling While Typing.

Leading off the explanation with "this is very similar to this other thing" isn't very useful to a user who has alt-clicked a type to find out what it is. It's worth noting the similarity to AsyncOpen, but the docs should start by explaining what this is and what is does.

Reply:
Thanks for the tip